src/groupeddataframe/groupeddataframe.jl

```diff
@@ -387,6 +391,29 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function _dict_to_tuple(key::AbstractDict{<:AbstractString} , gd)
+    T = keytype(key)
+    return ntuple(length(gd.cols)) do i 
+        # Most `<: AbstractString` types do not support constructors from symbols
+        key[T(string(gd.cols[i]))]
+    end
+end
+
+function _dict_to_tuple(key::AbstractDict{Symbol}, gd)
+    T = keytype(key)
+    return ntuple(length(gd.cols)) do i 
+        key[T(gd.cols[i])]
```

Comment:
here using `T` is not needed for sure, as `gd.cols` contains `Symbols`. Other comments as above